Service/SortTrait.php

```diff
@@ -111,7 +112,7 @@ private function loadSortsFromQuery(): void
     {
         $this->sortFields = [];
 
-        $request = $this->requestStack->getMasterRequest() ?? Request::createFromGlobals();
+        $request = RequestUtil::getMainRequest($this->requestStack) ?? Request::createFromGlobals();
```

Comment:
When creating a service for abstracting the request fetching, you can also move the `Request::createFromGlobals()` to the same service, because it is used in all places too.